### PR TITLE
Improve macOS Ci workflow

### PR DESCRIPTION
**Pull request by kelson42**
Many small improvements to increase readability, stability and speed.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -16,10 +16,7 @@ jobs:
       - name: Install packages
         run: |
           brew update
-          brew install pkg-config ninja || brew link --overwrite python
-
-      - name: Install python modules
-        run: pip3 install meson==0.49.2
+          brew install pkg-config ninja meson
```
Comment (veloman-yunkan) [inline .github/workflows/ci.yml:21]: This will install the latest version of `meson`. Is that OK?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -10,39 +10,45 @@ jobs:
...
       - name: Install packages
         run: |
           brew update
           brew install gcovr pkg-config ninja || brew link --overwrite python
+
       - name: Install python modules
         run: pip3 install meson==0.49.2 pytest
-      - name: Install deps
+
+      - name: Install dependences
```
Comment (veloman-yunkan) [inline .github/workflows/ci.yml:23]: dependences → dependencies

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kelson42) [reply .github/workflows/ci.yml:21]: yes, it should work

File: .github/workflows/ci.yml (hunk 2 above)
Comment (kelson42) [reply .github/workflows/ci.yml]: fixed

**Review by veloman-yunkan — APPROVED** [review]

**Pull request merged**